src/workerd/api/r2-bucket.c++

```diff
@@ -24,6 +24,25 @@
 #include <regex>
 
 namespace workerd::api::public_beta {
+kj::Own<kj::HttpClient> r2GetClient(IoContext& context,
+    uint subrequestChannel,
+    kj::LiteralStringConst op,
```

Comment:
I feel that defining a simple `struct UserTracing` or something would go a long way for readability above (due to `.method = "r2_delete"_kjc`) and below because of maybe casts?

Reply:
To make sure I understand correctly – I assume you mean a struct like this:
```
struct UserTracing {
  kj::LiteralStringConst op;
  kj::LiteralStringConst method;
  kj::Maybe<kj::StringPtr> bucket;
  kj::Maybe<StringTagParams> extraTag;
};
```

Then we could pass parameters simply as `{"r2_get"_kjc, "GetObject"_kjc, this->adminBucketName()}`, but you prefer using this instead?
```
{
.op = "r2_get"_kjc,
.method = "GetObject"_kjc,
.bucket = this->adminBucketName()
}
```

FWIW I was able to clean up the Maybe\<StringTagParams> wrappers in the last push.

Reply:
I would prefer the version with names spelled out, yes. It is much easier to read and understand how `r2_get` is different from `GetObject`